city_metrix/metrics/canopy_area_per_resident.py

```diff
@@ -45,8 +46,11 @@ def get_metric(self,
         if isinstance(tree_canopy_height_count, pd.DataFrame):
             result = tree_canopy_height_count.copy()
             result['value'] = tree_canopy_height_count['value'] / world_pop_sum['value']
+            result['value'] = result['value'].replace([np.inf, -np.inf], np.nan)
         else:
             result = tree_canopy_height_count / world_pop_sum
+            if np.isinf(result):
+                result = np.nan
```

Comment:
Is there a reason not to use `result = result.replace([np.inf, -np.inf], np.nan)` here?

Reply:
This is used for when the `result` is a number, and cannot apply the `.replace()` to a float. 

Or is the `result` a series? I haven't got a result that is not a dataframe in testing.

Reply:
Ya, I think this indicates that it should only be a dataframe or a series, but I am not sure when that would happen.

```
def get_metric(self,
                 geo_zone: GeoZone,
                 spatial_resolution:int = None) -> Union[pd.DataFrame | pd.Series]:
```